packages/reactivity/src/effectScope.ts

```diff
@@ -147,7 +149,16 @@ export function getCurrentScope() {
  */
 export function onScopeDispose(fn: () => void) {
   if (activeEffectScope) {
-    activeEffectScope.cleanups.push(fn)
+    activeEffectScope.cleanups.push(() => {
+      try {
+        fn()
+      } catch (e) {
+        console.error(
+          'An error occurred while executing a cleanup function:',
+          e,
+        )
```

Comment:
This isn't what I had in mind when I suggested handling the error cases. I just meant putting the `resetTracking()` call in a `finally` block to ensure it always gets called. Perhaps there are more complete ways to handle errors, but that feels like it's out of scope for the tracking changes being proposed in this PR.

There are a couple of problems with the error handling being proposed here:

1. This error message would be included in production builds, impacting bundle size. Error messages generally aren't included in production builds.
2. Catching the errors will prevent them being handled by `app.config.errorHandler`.

For example, consider this Playground:

- [Playground - `main`](https://play.vuejs.org/#eNp9UsFu2zAM/RVCO9gBAuWwnQon2NYF2HbYirZHXVyVSZTakiBRTYAg/15KrpO0KHowIL33SL9H6iD61li5jeJKmN67QHAAHbAl/OE9HGEVXA/Vc8JK2VdBJgZYzlrv5UAqq52NBIzA/Nyi5m+S2axkxcqsJYbgwu/WPnYYWFtLKduwjhOYL+CgLEDu5DqUnVvX1TKrYTPIqykUqbLHsWnvkqW6+sLnaiKm4tUSB2qiDsYTRKTkFyf/B8DVCjXdaedxCs6Wwy8TvYv4LvKYKmYJe72orNlEgWVItq7P7t82vGQAaBPcDizuoMSqK8JIFXcCOOZQ/DWzwTYb5gth7zseJd8AmodE5Cx8153RT3Mlhv9Hcp7diMV1hpvZoOKKZnYq58EQq8v8t9FZHk+xpIR2vTc82v+eDGdV4mo0q0TbdW73t2AUEk5HXG9QP32Ab+M+Y0rcBIwYnlGJE0e8OKSBXt79wz2fT2TvHlPH6k/IW+Q3kbLHQfYzlRdxoStu/5QlG7u+j8s9oY1jqGy0zLnoleAFX38S/Wz3q/xW6ng94vgCmoITDQ==)

Notice that clicking the button triggers the `errorHandler`. Contrast that with how it behaves with this PR:

- [Playground - this PR](https://deploy-preview-7265--vue-sfc-playground.netlify.app/#eNp9UsFu2zAM/RVCO9gBAuWwnQon2NYF2HbYirZHXVyVSZTakiBRTYAg/15KrpO0KHowIL33SL9H6iD61li5jeJKmN67QHAAHbAl/OE9HGEVXA/Vc8JK2VdBJgZYzlrv5UAqq52NBIzA/Nyi5m+S2axkxcqsJYbgwu/WPnYYWFtLKduwjhOYL+CgLEDu5DqUnVvX1TKrYTPIqykUqbLHsWnvkqW6+sLnaiKm4tUSB2qiDsYTRKTkFyf/B8DVCjXdaedxCs6Wwy8TvYv4LvKYKmYJe72orNlEgWVItq7P7t82vGQAaBPcDizuoMSqK8JIFXcCOOZQ/DWzwTYb5gth7zseJd8AmodE5Cx8153RT3Mlhv9Hcp7diMV1hpvZoOKKZnYq58EQq8v8t9FZHk+xpIR2vTc82v+eDGdV4mo0q0TbdW73t2AUEk5HXG9QP32Ab+M+Y0rcBIwYnlGJE0e8OKSBXt79wz2fT2TvHlPH6k/IW+Q3kbLHQfYzlRdxoStu/5QlG7u+j8s9oY1jqGy0zLnoleAFX38S/Wz3q/xW6ng94vgCmoITDQ==)

I don't think this is desirable.

Reply:
I agree with @skirtles-code.